File: tests/python/cases/case_010_publish_ledger.py
"""
Task case 010: publish ledger payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_010_publish_ledger(input_data):
    # Task: publish ledger records and prepare transport-ready payload.
    task_name = 'publish_ledger'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'ledger',
        'stage': 'publish',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # SAFE: payload remains local; only primitive checksum string is returned.
    return payload['checksum']

File: tests/nodejs/cases/case_010_publish_ledger.js
/**
 * Task case 010: publish_ledger
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case010PublishLedger(input) {
  // Task: publish ledger records and prepare transport-ready payload.
  const taskName = 'publish_ledger';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'ledger',
    stage: 'publish',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // SAFE: payload remains local; only primitive checksum string is returned.
  return payload.checksum;
}

module.exports = {
  case010PublishLedger,
};
